vendor/brave-ios/BATBraveRewards.m

```diff
@@ -108,18 +108,49 @@ - (void)setupLedgerAndAds
 
 @implementation BATBraveRewards (Reporting)
 
-- (void)reportTabUpdated:(NSInteger)tabId url:(NSURL *)url isSelected:(BOOL)isSelected isPrivate:(BOOL)isPrivate
+- (void)reportTabUpdated:(NSInteger)tabId
+                     url:(NSURL *)url
+              isSelected:(BOOL)isSelected
+               isPrivate:(BOOL)isPrivate
+              completion:(void (^)(BOOL verifiedPublisher))completion
 {
   if (isSelected) {
     self.ledger.selectedTabId = (UInt32)tabId;
   }
+
+  [self onTabRetrieved:tabId url:url];
   [self.ads reportTabUpdated:tabId url:url isSelected:isSelected isPrivate:isPrivate];
+  
+  if (tabId == self.ledger.selectedTabId) {
+    [self.ledger publisherInfoForId:url.host completion:^(BATPublisherInfo * _Nullable info) {
+      completion(info.verified);
+    }];
+  }
 }
 
-- (void)reportLoadedPageWithURL:(NSURL *)url tabId:(UInt32)tabId html:(NSString *)html
+- (void)reportLoadedPageWithURL:(NSURL *)url
+                          tabId:(UInt32)tabId
+                           html:(NSString *)html
+                     completion:(void (^)(BOOL verifiedPublisher))completion
 {
+  [self onTabRetrieved:tabId url:url];
   [self.ads reportLoadedPageWithURL:url html:html];
   [self.ledger reportLoadedPageWithURL:url tabId:tabId];
+  
+  if (tabId == self.ledger.selectedTabId) {
```

Comment:
Sorry, I meant only the `reportTabUpdated` one specific to selected, since we'd want to create the publisher for all pages, but we may not care about updating the badge for verified publisher on a tab that isn't selected.